rust/agama-lib/src/error.rs

```diff
@@ -18,6 +18,8 @@ pub enum ServiceError {
     // specific error will be printed too
     #[error("Error: {0}")]
     Anyhow(#[from] anyhow::Error),
+    #[error("Network client error: '{0}'")]
+    NetworkClientError(String),
```

Comment:
A little bit generic, isn't it? Let's keep it as it is but...

Reply:
Well, thinking again about this, I would say it is wrong. We are just throwing away the cause of the error for no good reason. Please, have a look to the [Error trait](https://doc.rust-lang.org/std/error/trait.Error.html) definition.